src/GitHub.Exports/Models/SimpleRepositoryModel.cs

```diff
@@ -133,39 +144,50 @@ public string HeadSha
         {
             get
             {
-                var repo = GitService.GitServiceHelper.GetRepo(LocalPath);
+                var repo = GitService.GitServiceHelper.GetRepository(LocalPath);
                 return repo?.Commits.FirstOrDefault()?.Sha ?? String.Empty;
             }
         }
 
+        public IBranch CurrentBranch
+        {
+            get
+            {
+                var repo = GitService.GitServiceHelper.GetRepository(LocalPath);
+                return new BranchModel(repo?.Head, this);
+            }
+        }
+
+
         /// <summary>
         /// Note: We don't consider CloneUrl a part of the hash code because it can change during the lifetime
         /// of a repository. Equals takes care of any hash collisions because of this
         /// </summary>
         /// <returns></returns>
         public override int GetHashCode()
         {
-            return (Name?.GetHashCode() ?? 0) ^ (LocalPath?.TrimEnd('\\').ToUpperInvariant().GetHashCode() ?? 0);
+            return 17 * 23 + (Name?.GetHashCode() ?? 0) * 23 + (Owner?.GetHashCode() ?? 0) * 23 + (LocalPath?.TrimEnd('\\').ToUpperInvariant().GetHashCode() ?? 0);
         }
 
         public override bool Equals(object obj)
         {
             if (ReferenceEquals(this, obj))
                 return true;
             var other = obj as SimpleRepositoryModel;
-            return other != null && String.Equals(Name, other.Name) && String.Equals(CloneUrl, other.CloneUrl) && String.Equals(LocalPath?.TrimEnd('\\'), other.LocalPath?.TrimEnd('\\'), StringComparison.CurrentCultureIgnoreCase);
+            return other != null && String.Equals(Name, other.Name) && String.Equals(Owner, other.Owner) && String.Equals(CloneUrl, other.CloneUrl) && String.Equals(LocalPath?.TrimEnd('\\'), other.LocalPath?.TrimEnd('\\'), StringComparison.CurrentCultureIgnoreCase);
         }
 
         bool IEquatable<SimpleRepositoryModel>.Equals(SimpleRepositoryModel other)
         {
             if (ReferenceEquals(this, other))
                 return true;
-            return other != null && String.Equals(Name, other.Name) && String.Equals(CloneUrl, other.CloneUrl) && String.Equals(LocalPath?.TrimEnd('\\'), other.LocalPath?.TrimEnd('\\'), StringComparison.CurrentCultureIgnoreCase);
+            return other != null && String.Equals(Name, other.Name) && String.Equals(Owner, other.Owner) && String.Equals(CloneUrl, other.CloneUrl) && String.Equals(LocalPath?.TrimEnd('\\'), other.LocalPath?.TrimEnd('\\'), StringComparison.CurrentCultureIgnoreCase);
```

Comment:
Kinda worries me to see this long line copy/pasted in 2 places? Could you make `Equals(SimpleRepositoryModel)` a normal method (not an explicit interface method) and call `Equals(SimpleRepositoryModel)` from `Equals(object)`?